src/gpio.rs

```diff
@@ -355,6 +525,29 @@ macro_rules! impl_input_base {
             }
         }
 
+        #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "alloc"))]
+        impl $pxi<Input> {
+            /// # Safety
+            ///
+            /// The callback passed to this method is executed in the context of an
+            /// interrupt handler. So you should take care of what is done in it.
+            pub unsafe fn into_subscribed(
+                mut self,
```

Comment:
Not sure I understand something here. Is it possible to subscribe, unsubscribe, then subscribe again? The PR description implies it should be possible.

I think the above function consumes the pin so we can't subscribe twice. But in which case how do we subscribe again if we have unsubscribed?

Apologies if this is obvious to everyone except me :-)

Reply:
No problem. You can check [this example usage](https://github.com/pyaillet/esp32-isr/blob/use-patched-eih/src/main.rs#L68).
When you want to register your callback, you get a `GpioPin<SubscribedInput>` as a result. If you want to reuse the pin/unregister your handler, you just have to change the state of the pin by using a `into_*` method.